File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/analysis/acceptance-project/src/main/java/surveycrawler/RetreatCommand.java
package surveycrawler;

public class RetreatCommand extends CrawlerMovement {

    // testing

    @Override
    public boolean isFor(char aCharacter) {
        return aCharacter == 't';
    }

    // executing

    @Override
    public void executeOn(SurveyCrawler aCrawler) {
        aCrawler.retreat();
    }

    // undoing

    @Override
    public CrawlerMovement inverse() {
        return new AdvanceCommand();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/analysis/own-tests-project/src/surveycrawler/retreat_command.py
from surveycrawler.command import Command


class RetreatCommand(Command):

    # character

    @classmethod
    def character(cls):
        return 't'

    # executing

    def execute_on(self, a_crawler):
        a_crawler.retreat()

    def undo_on(self, a_crawler):
        a_crawler.move_one_cell_forward()
